src/main/scala/firrtl/passes/InferWidths.scala

```diff
@@ -81,7 +93,7 @@ object InferWidths extends Pass {
 
     def remove_cycle(n: String)(w: Width): Width = {
       //;println-all-debug(["Removing cycle for " n " inside " w])
-      w map remove_cycle(n) match {
```

Comment:
I think the entire width inference needs to be better documented, and I think that's a separate PR. I'll submit an issue for it.